src/commands/compose/getComposeServiceList.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { IActionContext, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { execAsync } from '../../utils/spawnAsync';
+
+const composeCommandReplaceRegex = /(up|down).*$/i;
+
+export async function getComposeServiceList(context: IActionContext, workspaceFolder: vscode.WorkspaceFolder, composeCommand: string): Promise<string> {
+    const services = await getServices(workspaceFolder, composeCommand);
+
+    // Fetch the previously chosen services list. By default, all will be selected.
+    const workspaceServiceListKey = `vscode-docker.composeServices.${workspaceFolder.name}`;
+    const previousChoices = ext.context.workspaceState.get<string[]>(workspaceServiceListKey, services);
+
+    const pickChoices: IAzureQuickPickItem<string>[] = services.map(s => {

```

Comment:
Nit: when returning objects, the `return` and braces can be omitted by using parentheses.
```javascript
s => ({
    label: s,
    data: s,
    picked: ...
}));
```

Reply:
So THAT's how to do it!